### DISCUSSION TITLE
Broh5: Browser-based GUI HDF Viewer

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#1941] [Show and tell]
Would like to share my project using Nicegui. It's a viewer for hdf/h5/nxs files and installable using pip.
`pip install broh5` 

![fig_01](https://github.com/nghia-vo/broh5/raw/main/figs/fig_01.png)

Video demo:

[![Demo](https://img.youtube.com/vi/lEJ6LKOaIFk/sddefault.jpg)](https://www.youtube.com/watch?v=lEJ6LKOaIFk)

Source code: https://github.com/nghia-vo/broh5

The codebase is designed using the RUI (Rendering-Utilities-Interactions) concept, which is known as the MVC (Model-View-Controller) pattern in the GUI development community. The name 'MVC' may not be very intuitive for those new to GUI development. However, this design allows for the development of complex software and makes it easier to extend its capabilities. To make things easier for beginners, a monolithic approach to the codebase is also provided for comparison.

Big thanks to the Nicegui team: @falkoschindler @rodja  to make such a fantastic software.

**Comment by rodja** [comment]
Wow. Very cool. Thanks for sharing @nghia-vo. I've added it to the list of community projects in our wiki: https://github.com/zauberzeug/nicegui/wiki#community-projects
